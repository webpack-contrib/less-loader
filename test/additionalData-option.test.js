import {
  compile,
  getCodeFromBundle,
  getCodeFromLess,
  getCompiler,
  getErrors,
  getWarnings,
} from "./helpers";

describe('"additionalData" option', () => {
  it("should work as string", async () => {
    const testId = "./additional-data.less";
    const additionalData = "@background: coral;";
    const compiler = getCompiler(testId, { additionalData });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId, { additionalData });

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should as function", async () => {
    const testId = "./additional-data.less";
    const additionalData = (content, loaderContext) => {
      const { resourcePath, rootContext } = loaderContext;
      // eslint-disable-next-line global-require
      const relativePath = require("path").relative(rootContext, resourcePath);

      return `
          /* RelativePath: ${relativePath}; */

          @background: coral;
          ${content};
          .custom-class {color: red};
        `;
    };
    const compiler = getCompiler(testId, { additionalData });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId, { additionalData });

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should work as async function", async () => {
    const testId = "./additional-data.less";
    const additionalData = async (content, loaderContext) => {
      const { resourcePath, rootContext } = loaderContext;
      // eslint-disable-next-line global-require
      const relativePath = require("path").relative(rootContext, resourcePath);

      return `
          /* RelativePath: ${relativePath}; */

          @background: coral;
          ${content};
          .custom-class {color: red};
        `;
    };
    const compiler = getCompiler(testId, { additionalData });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId, { additionalData });

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
