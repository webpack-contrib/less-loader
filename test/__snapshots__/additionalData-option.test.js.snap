// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"additionalData" option should as function: css 1`] = `
"/* RelativePath: additional-data.less; */
.background {
  color: coral;
}
.custom-class {
  color: red;
}
"
`;

exports[`"additionalData" option should as function: errors 1`] = `[]`;

exports[`"additionalData" option should as function: warnings 1`] = `[]`;

exports[`"additionalData" option should work as async function: css 1`] = `
"/* RelativePath: additional-data.less; */
.background {
  color: coral;
}
.custom-class {
  color: red;
}
"
`;

exports[`"additionalData" option should work as async function: errors 1`] = `[]`;

exports[`"additionalData" option should work as async function: warnings 1`] = `[]`;

exports[`"additionalData" option should work as string: css 1`] = `
".background {
  color: coral;
}
"
`;

exports[`"additionalData" option should work as string: errors 1`] = `[]`;

exports[`"additionalData" option should work as string: warnings 1`] = `[]`;
